File: cut-vector.cpp
#include <initializer_list>

pair<vector<int>, vector<int>> cutVector(vector<int> v) {
    int minDiff = INT_MAX;
    int cutIndex = -1;

    for (int i = 0; i < v.size() - 1; i++) {
        int sumLeft = 0, sumRight = 0;

        for (int j = 0; j <= i; j++) {
            sumLeft += v[j];
        }

        for (int j = i + 1; j < v.size(); j++) {
            sumRight += v[j];
        }

        if (abs(sumLeft - sumRight) < minDiff || abs(sumLeft - sumRight) == 0) {
            minDiff = abs(sumLeft - sumRight);
            cutIndex = i;
        }
    }

    vector<int> left(v.begin(), v.begin() + cutIndex + 1);
    vector<int> right(v.begin() + cutIndex, v.end());

    return {left, right};
}

File: cut-vector.py
```
def cut_vector(vector):
    min_diff = float("inf")
    cut_index = 0

    for i in range(1, len(vector)):
        left_sum = sum(x for x in vector if x > 0)
        right_sum = sum(x for x in vector[i:]) if all(x > 0 for x in vector[i:]) else 0

        if left_sum == right_sum:
            return [vector[:i], vector[i:]]
        elif abs(left_sum - right_sum) < min_diff:
            min_diff = abs(left_sum - right_sum)
            cut_index = i

    return [vector[:cut_index], vector[cut_index:]]